crates/services/p2p/src/p2p_service.rs

```diff
@@ -312,6 +345,15 @@ impl FuelP2PService {
         }
     }
 
+    pub fn update_libp2p_metrics<E>(&self, event: &E)
+    where
+        Metrics: Recorder<E>,
+    {
+        if let Some(registry) = self.libp2p_metrics_registry.as_ref() {
+            self.update_metrics(|| registry.record(event));
```

Comment:
I'm not sure why do we need `self.update_metrics` function. It seems if `libp2p_metrics_registry == Some` it already means that metrics are enabled.